leader-election-impl/src/main/java/com/palantir/leader/LeadershipObserver.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.leader;
+
+public interface LeadershipObserver {
```

Comment:
I don't really like being able to change the list of tasks during the lifecycle of this object.  I'd rather the `Runnable`s were passed in at construction time and then couldn't be changed.

Reply:
It is a bit hard to implement as construction of the observer and attaching the tasks takes place on different levels. I changed setters in observer to be synchronized, and also used a synchronized list to keep `Runnable`s to avoid race conditions.